pkg/tests/ossm/smcp_annotation_test.go

```diff
@@ -69,29 +69,43 @@ func TestSMCPAnnotations(t *testing.T) {
 			oc.WaitDeploymentRolloutComplete(t, ns, "testenv")
 
 			t.LogStep("Get annotations and verify that the pod has the expected: test1.annotation-from-smcp : test1, test2.annotation-from-smcp : [\"test2\"], test3.annotation-from-smcp : {test3}")
-			annotations := GetPodAnnotations(t, pod.MatchingSelector("app=env", ns))
-			assertAnnotationIsPresent(t, annotations, "test1.annotation-from-smcp", "test1")
-			assertAnnotationIsPresent(t, annotations, "test2.annotation-from-smcp", `["test2"]`)
-			assertAnnotationIsPresent(t, annotations, "test3.annotation-from-smcp", "{test3}")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				annotations := VerifyAndGetPodAnnotation(t, pod.MatchingSelector("app=env", ns))
+				assertAnnotationIsPresent(t, pod.MatchingSelector("app=env", ns), annotations, "test1.annotation-from-smcp", "test1")
+				assertAnnotationIsPresent(t, pod.MatchingSelector("app=env", ns), annotations, "test2.annotation-from-smcp", `["test2"]`)
+				assertAnnotationIsPresent(t, pod.MatchingSelector("app=env", ns), annotations, "test3.annotation-from-smcp", "{test3}")
+			})
 		})
 	})
 }
 
-func GetPodAnnotations(t TestHelper, podLocator oc.PodLocatorFunc) map[string]string {
-	annotations := map[string]string{}
+func VerifyAndGetPodAnnotation(t TestHelper, podLocator oc.PodLocatorFunc) map[string]string {
+	var data map[string]interface{}
 	po := podLocator(t, oc.DefaultOC)
-	retry.UntilSuccess(t, func(t test.TestHelper) {
-		output := shell.Executef(t, "kubectl get pod %s -n %s -o jsonpath='{.metadata.annotations}'", po.Name, po.Namespace)
-		err := json.Unmarshal([]byte(output), &annotations)
-		if err != nil {
-			t.Fatalf("Error parsing pod annotations json: %v", err)
-		}
-	})
+	yamlString := oc.GetYaml(t, po.Namespace, "pod", po.Name)
```

Comment:
Instead of getting pod, you could get pod annotations with the command below. Then you will no longer need to parse YAML structure, because that command will return a string.
```
oc get pods <pod-name> -o jsonpath='{.metadata.annotations.sidecar\.maistra\.io/proxyEnv}'
```

Reply:
I don't understand that point of view... Why do we might prefer parsing whole YAML instead of getting what we need. It does not make sense for me.

Reply:
This is the comment from Marko: https://github.com/maistra/maistra-test-tool/pull/544#discussion_r1190836485

Reply:
I think it's not more easier but it's cleaner for test reading purposes.

Reply:
No offence, but YAML parsers are always terrible. I really can't understand why we can't use `kubectl ... jsonpath`. It's just strange to me that we have powerful tools, but we implement them from scratch. It does not make sense.
I'm sorry that I didn't give you an approval yet, but I also work with this code and I don't want to read, debug or modify these functions.

Reply:
No problem, I understand your point of view. I think the thing here is trying to make the code readable for anyone, even non-developer, so all these functions are wrapped also to be reusable. Yes we can use kubectl or oc to get the information or execute commands directly but with these functions created we can reuse the code in some many places avoiding having to always write test cases with a lot of steps because we are using in each one a kubectl or oc command

Reply:
@luksa Hi, please can you take a look at this discussion?

Reply:
There's a reason why we have `oc.Xyz()` functions instead of using `shell.Execute` throughout the entire codebase. That way, if there's a change we need to make to all the oc functions (like the change where we had to introduce `withKubeconfig`), we can make that change quickly and without any special effor, while ensuring that the change doesn't break anything. 

If we pollute the code with a lot of custom `shell.Execute('oc ...')` calls, making those types of changes will be difficult, take a lot of time, and likely result in bugs that will only show up later. That's why I suggested using `oc.GetYaml()`. However, I wasn't saying that you must use `oc.GetYaml()`; you could have used an alternative. For example, you could have introduce `oc.GetYamlPart(..., jsonPath string)` which only returns the part of the YAML you want. Or even introduce `oc.GetAnnotations()`. In both of these cases, any changes that we'd need to make in the future, would be easy to do. Plus, you'd have a common function that would probably end up being used  in other places as well. If you compare this to using `shell.Execute('oc get...')` in the test, you see that using `oc.GetYaml`, or introducing `oc.GetYamlPart/oc.getAnnotations` is a much better solution. 

We have to be mindful of introducing unnecessary technical debt. That's what got us into the maistra-test-tool mess in the first place. If you don't don't care about code quality, we'll soon be in the same mess again.

Reply:
>  If you don't don't care about code quality, we'll soon be in the same mess again.

It's always a subjective opinion. Writing custom yaml parsers does not seem to be a direction for god quality code **from my point of view**. But I won't block this PR, it's up to you.

Reply:
I never said to write a custom yaml parser :). The only issue I had was with calling `oc` in a generic `shell.Execute`.

Reply:
This can be improved in another PR, because I'm creating a new func oc.GetJson that allows to the user get data from JsonPath so for this example: `oc get pods <pod-name> -o jsonpath='{.metadata.annotations.sidecar\.maistra\.io/proxyEnv}'`

We will used like this:
`oc.GetJson(t, ns, "pod", "pod-name", "{.metadata.annotations.sidecar\.maistra\.io/proxyEnv}")`

And we will get the string from that path of the json